src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        switch (size) {
+            case 0 :
+                first = new Node<>(null, value, null);
+                last = first;
+                break;
+            case 1 :
+                last = new Node<>(first, value, null);
+                first.next = last;
+                break;
+            default :
+                last.next = new Node<>(last, value, null);
+                last = last.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToBounds(index, size);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNodeByIndex(index);
+        node.prev = new Node<>(node.prev, value, node);
+        if (node.prev.prev == null) {
+            first = node.prev;
+        } else {
+            node.prev.prev.next = node.prev;
+        }
+        size++;
```

Comment:
```suggestion
    Node<T> next = getNodeByIndex(index);
    Node<T> prev = next.prev;
    Node<T> newNode = new Node<>(prev, value, next);
    if (prev == null) {
      first = newNode;
    } else {
      prev.next = newNode;
    }
    next.prev = newNode;
    size++;
```

It's hard to read smth like `node.prev.prev.next`